### PR TITLE
[JENKINS-35452] A step towards removing more 'slave' terminology.

### PR DESCRIPTION
**Pull request by jeffret-b**
Overload the mechanisms for obtaining the jenkins-agent.jnlp (formerly slave-agent.jnlp) file for launching inbound TCP agents.
Serve the file at either of the two locations.
Change the UI (etc.) documentation to reference the correct term / file.

This doesn't allow us to remove the old, incorrect term, but allows us a path forward. Existing agents and configurations continue to work. New ones are suggested to use the new terms and locations. This obsoletes the old. At some point we can remove the old and anyone who hasn't yet done the replacement on their configuration will have to do it. Anyone interested can do it immediately as it doesn't require any agent (Remoting) changes. 

See [JENKINS-35452](https://issues.jenkins-ci.org/browse/JENKINS-35452).

### Proposed changelog entries

* Entry 1: [JENKINS-35452], Change the standard URL for obtaining the Inbound Agent configuration file from "[...]/slave-agent.jnlp" to "[...]/jenkins-agent.jnlp". The old name is obsoleted and will be removed at a future time.
* ...

### Proposed upgrade guidelines

[May want to insert a recommendation on changing the file name for existing agents.]

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/Functions.java
Comment (daniel-beck) [inline core/src/main/java/hudson/Functions.java:1877]: https://github.com/search?q=org%3Ajenkinsci+%22slave-agent.jnlp%22&type=code looks like we could get away with just renaming the existing file instead of duplicating it.

If we need to keep the old file for some reason, would just have that `st:include` the new one if possible to cut down duplication.

Comment (jeffret-b) [reply core/src/main/java/hudson/Functions.java:1877]: Thanks for catching that. It was supposed to be removed but while I was getting things working it crept back in. Removing ....

**Review by jvz — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
It would be nice to keep tests for the old URL, it will take some time to clean it up

**Comment by jeffret-b** [comment]
Added tests for the new and old URL.

**Review by res0nance — APPROVED** [review]

**Review by markjacksonfishing — APPROVED** [review]

**Comment by MRamonLeon** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/java/hudson/slaves/SlaveComputer.java
Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/SlaveComputer.java:821]: Unnecessary code duplication as `@WebMethod` takes a list of names.

If this is the future preferable API for this (even if unseen), should mark the old one `@deprecated`.

Also missing `@since` or `@Restricted`.

File: core/src/main/java/hudson/slaves/SlaveComputer.java
Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/SlaveComputer.java:827]: As above.

**Pull request merged**
